### PR TITLE
Fix: update metrics collection banner to modal with management toggle settings

### PR DESCRIPTION
**Pull request by djmcquillan**
fixes: #340 

Updated PR for #340 adding modal and toggles for metrics collection

https://user-images.githubusercontent.com/10619673/212779585-fd8b5e3b-2614-4136-814d-437f7d12de7a.mov

### CODE REVIEW

**Review by SgtPooki — APPROVED** [review]

**Comment by juliaxbow** [comment]
Not blockers but some comments included in the screenshots below. Also there's an open question in the ignite channel about whether the metrics policy should link out [to the github isse
](https://github.com/ipfs/ipfs-gui/issues/125) or the [companion privacy policy ](https://github.com/ipfs/ipfs-companion/blob/main/PRIVACY-POLICY.md). I believe the last decision was to move towards the former while we wait on a broader-reaching policy from Legal. 

### Desktop
#### Current
Screen Shot 2023-01-17 at 1 29 55 PM

#### With suggested changes
Screen Shot 2023-01-17 at 1 33 05 PM

### Mobile:
Screen Shot 2023-01-17 at 1 35 22 PM

File: src/metrics.ts [hunk 1]
```diff
@@ -102,17 +120,10 @@ function loadCountly (): void {
...
   if (metricsConsent != null) {
-    try {
-      addConsent(JSON.parse(metricsConsent))
-    } catch {
-      displayConsentBanner()
-    }
+    addConsent(JSON.parse(metricsConsent))
   } else {
-    displayConsentBanner()
+    addConsent(['all'])
```
Comment (SgtPooki) [inline src/metrics.ts:125]: this should be addConsent(['minimal']) shouldn't it

**Review by SgtPooki — CHANGES_REQUESTED** [review]

**Comment by SgtPooki** [comment]
> Also there's an open question in the ignite channel about whether the metrics policy should link out https://github.com/ipfs/ipfs-gui/issues/125
or the [companion privacy policy ](https://github.com/ipfs/ipfs-companion/blob/main/PRIVACY-POLICY.md). I believe the last decision was to move towards the former while we wait on a broader-reaching policy from Legal.

We should link to https://github.com/ipfs-shipyard/ignite-metrics/blob/main/docs/telemetry/COLLECTION_POLICY.md

File: src/index.html [hunk 2]
```diff
@@ -60,41 +60,64 @@ <h1 class='f3 fw2 montserrat aqua ttu ma0'>Public Gateways</h1>
...
+          Metrics and Telemetry Settings
+        </h3>
+        <!-- manage preferences -->
+        <span class="metrics-notification-text montserrat">
+          This site collects metrics and other telemetry data in order to improve and prioritize work on IPFS. It does not collect any personally identifiable information.
+        </span>
+        <div class="metrics-notification-customize">
+          <span class="metrics-notification-text metrics-notification-customize-text montserrat">
+            Opt-in or out at any time by using the toggle below.
+            For more details, read the full<a href="https://github.com/ipfs/ipfs-gui/issues/125"> metrics collection policy</a>.
```
Comment (SgtPooki) [inline src/index.html:91]: we should link to https://github.com/ipfs-shipyard/ignite-metrics/blob/main/docs/telemetry/COLLECTION_POLICY.md

```suggestion
            For more details, read the full<a href="https://github.com/ipfs-shipyard/ignite-metrics/blob/main/docs/telemetry/COLLECTION_POLICY.md"> metrics collection policy</a>.
```

**Review by SgtPooki — APPROVED** [review]
Lgtm, lets get this shipped

**Pull request merged**